pyo3-macros-backend/src/pyclass.rs

```diff
@@ -1783,6 +1800,33 @@ fn pyclass_richcmp(
     }
 }
 
+fn pyclass_hash(
+    options: &PyClassPyO3Options,
+    cls: &syn::Type,
+    ctx: &Ctx,
+) -> Result<(Option<syn::ImplItemFn>, Option<MethodAndSlotDef>)> {
+    if options.hash.is_some() {
+        ensure_spanned!(options.frozen.is_some(), options.hash.span() => "The `hash` option requires the `frozen` option.");
+        ensure_spanned!(options.eq.is_some(), options.hash.span() => "The `hash` option requires the `eq` option.");
```

Comment:
For these error messages I think it's quite likely users might hit the `frozen` error, fix that, and then hit the `eq` error. If you agree, it might be nice if we also special-case if both are missing and emit an error message requesting both options so users can fix in a single compile. 

Or we could keep it simple and just always check for both in one `ensure_spanned` and just emit the combined message.

Reply:
I agree, I added a variant to the `ensure_spanned` macro that evaluates multiple branches and returns all errors.